### PR TITLE
add more key bindings for Rust major mode

### PR DESCRIPTION
**Pull request by o8vm**
I've added key bindings for `+Actions`, `+Goto`, and `+Peek`, referring
to spacemacs LSP layer [docs](https://develop.spacemacs.org/layers/+tools/lsp/README.html) and VSpaceCode [conventions](https://vspacecode.github.io/docs/conventions/).

I have not added bindings starting with `SPC m c`. To implement these bindings of [Rust layer](https://develop.spacemacs.org/layers/+lang/rust/README.html), I think we will need to implement a new extension using `cargo`.

### CODE REVIEW

**Comment by stevenguh** [comment]
LGTM.

- Do you mind updating the CHANGLOG?
- Have you tested all the bindings on rust lang? I know some language didn't implement all the LSP functions like for example the call hierarchy. Just want to make sure they all work.

File: package.json [hunk 1]
```diff
@@ -4829,6 +4880,99 @@
...
+													{
+														"key": "d",
+														"name": "Find defenition",
+														"icon": "symbol-function",
+														"type": "command",
+														"command": "editor.action.revealDefinition"
+													},
+													{
+														"key": "h",
+														"name": "Go to call hierachy",
```
Comment (stevenguh) [inline package.json:4899]:
```suggestion
														"name": "Go to call hierarchy",
```

**Comment by o8vm** [comment]
> Do you mind updating the CHANGLOG

I'll update Changelog later.

> Have you tested all the bindings on rust lang? I know some language didn't implement all the LSP functions like for example the call hierarchy. Just want to make sure they all work.

Yes. I've confirmed that everything works. I've not implemented anything that does not work.

File: package.json [hunk 2]
```diff
@@ -4829,6 +4880,99 @@
...
+													{
+														"key": "d",
+														"name": "Peek definition",
+														"icon": "symbol-function",
+														"type": "command",
+														"command": "editor.action.peekDefinition"
+													},
+													{
+														"key": "h",
+														"name": "show call hierarchy",
```
Comment (stevenguh) [inline package.json:4956]:
```suggestion
														"name": "Show call hierarchy",
```

**Comment by stevenguh** [comment]
> Yes. I've confirmed that everything works. I've not implemented anything that does not work.

Awesome!!

---
@MarcoIeni I realized this bindings are different than some other major modes that we implemented especially in the `m g` and `m G`. This bindings are closer to the bindings in [LSP layer](https://develop.spacemacs.org/layers/+tools/lsp/README.html#core-key-bindings). **I feel like we should follow it to maintain compatibility.** What do you think?

If we would like to follow this bindings as convention:
- [] Update existing major modes
- [] Update convention page to reflect (see  https://github.com/VSpaceCode/vspacecode.github.io/pull/50)

Otherwise, this PR will need to follow the current unwritten convention.

**Comment by MarcoIeni** [comment]
I tested all the bindings and they all work.
I really like that they follow the LSP conventions of spacemacs. Great work!

**Review by MarcoIeni — APPROVED** [review]

**Comment by MarcoIeni** [comment]
>  I feel like we should follow it to maintain compatibility.

Steven, I understood that you want to follow the convention of this PR, i.e. the one of spacemacs. Is that right?

In this case I agree with you, so we should update the other major modes and update #50 accordingly.

**Comment by stevenguh** [comment]
> Steven, I understood that you want to follow the convention of this PR, i.e. the one of spacemacs. Is that right?
> 
> In this case I agree with you, so we should update the other major modes and update #50 accordingly.

That's correct. Added a tracking issue for updating existing bindings #239.

**Pull request merged**